File: fixtures/scale/scala/large/src/cryptobench/bench/Timer.scala
package cryptobench.bench

final class Timer {
  private var startedAt = 0L

  def start(): Unit = startedAt = System.nanoTime()

  def stop(): Long = System.nanoTime() - startedAt
}

object Timer {
  def toMillis(nanos: Long): Double = nanos / 1000000.0
}

File: fixtures/scale/java/large/src/cryptobench/bench/Timer.java
package cryptobench.bench;

public class Timer {
    private long startedAt;

    public void start() {
        startedAt = System.nanoTime();
    }

    public long stop() {
        return System.nanoTime() - startedAt;
    }

    public static double toMillis(long nanos) {
        return nanos / 1_000_000.0;
    }
}
